scripts/hooks/init.js

```diff
@@ -223,6 +223,21 @@ Hooks.once("init", () => {
         default: true,
         type: Boolean
       });
+
+      // Register default test difficulty
+      game.settings.register("wfrp4e", "testDefaultDifficulty", {
+        name: "SETTINGS.TestDialogDefaultDifficulty",
+        hint: "SETTINGS.TestDialogDefaultDifficultyHint",
+        scope: "world",
+        config: true,
+        default: "challenging",
+        type: String,
+        choices: {
+          "average" : "SETTINGS.TestDialogDefaultDifficultyAverage",
+          "challenging" : "SETTINGS.TestDialogDefaultDifficultyChallenging"
+        }
+        // type: Boolean
+      });
```

Comment:
Thanks for this - gives me food for thought. 

I did consider a **boolean** originally, then decided to go with some more explicit setting options. I'm easy either way tbh. I can see how the boolean version gives cleaner code. It also means I can strip out the language settings for the dropdown options. So, I'll make those changes. 👍 

I liked your hint but the title confused me. Alternatively: 
Title: "Default Non-Combat Difficulty" - because when you check it, the setting you're changing is the default outside of combat.  
Hint: "Sets default Test difficulty to Average if there is no Combat. Outside of combat, or if unchecked, default difficulty is Challenging. "

I mulled over using the **game.combat 'started'** state. If GMs set up encounter(s) in a scene, by placing tokens (which may be hidden) and then adding them to the tracker in anticipation of a showdown, that would trigger game.combat. So to allow for advance setup without breaking the 'waiting for stuff to go south' sitch, I went for making sure the combat had also started. 

AFAICT keeping the 'started' state shouldn't affect the precedence you've described, which I agree with. My testing shows that is honoured. 

> The line testDifficulty : income ? "average" : "challenging" is still deleted,

That **income check** line is still intended to be deleted, as the various combinations I tried in `actor-wfrp4e.js` also seemed to apply to non-income skill tests. The cleanest way I found was to add the income check to `prepareTest` instead (so it is still in play).  That always gave the right outcome, and overrides scene stress or default test difficulty based on system setting. In any case, user setting also takes precedence.

> 2. Specified by action (income, trait default)

I've not done anything special for trait default override. I think it's handled automatically by the above. I may have missed something?

Reply:
I'm still confused on the issues you faced with income, because it seems like it should work fine to me, maybe I just need to see for myself, but ideally my vision is:

In the `setupSkill` function,
`testDifficulty : income ? "average" : "challenging"` provides a pre-set difficulty. 

Afterwards, in `prepareTest()`, 
`testDifficulty: dialogOptions.data.testDifficulty || sceneStress`

This is exactly how it worked previously, except for the added `|| sceneStress`, is something wrong with this logic?

So, `dialogOptions.data.testDifficulty` takes precedence over scene stress, this is good and I'm harping on this because 1. I don't like income in `prepareTest()` just because prepareTest() should be very general, and it doesn't feel right in being there, and more importantly 2. this is also how Trait default difficulty works (which I don't think you actually have since you branched off stable, I would merge `dev-backend` into your branch and ensure everything still works, we want Trait default difficulty to override Scene stress.